### PR TITLE
testing: add yaml lint github action

### PR DESCRIPTION
**Pull request by grayside**
Add YAML Linting based on https://github.com/GoogleCloudPlatform/public-datasets-pipelines/blob/main/.github/workflows/yaml-lint.yaml

### CODE REVIEW

File: .yamllint.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+extends: default
+
+rules:
```
Comment (averikitsch) [inline .yamllint.yaml:17]: Based on https://developers.google.com/devsite/reference/metadata/yaml#nested_yaml_elements
I believe we should add:
```
  indentation:
    spaces: 2
    indent-sequences: false
```

File: src/connectors/connector_cloudbuild.workflows.yaml [hunk 2]
```diff
@@ -13,13 +13,15 @@
...
-# The workflow assumes a properly defined source tgz file already
-# exists in a Cloud Storage bucket: gs://your-project-id_cloudbuild/source/placeholder_src.tgz 
-# and a repository "your-project-id-docker-repo" already
-# exists in Artifact Registry where the image will be stored.
-# The new table and dataset are both deleted in the following steps.
+# This workflow demonstrates how to use the Cloud Build connector
+# to start a build that creates and pushes a conainer image to
+# an Artifact Registry repository.
+#
+# Prerequisites:
```
Comment (averikitsch) [inline src/connectors/connector_cloudbuild.workflows.yaml:20]: This is great!

File: .github/workflows/yaml_check.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+name: Lint YAML
+on: [pull_request]
+jobs:
+  build:
```
Comment (averikitsch) [inline .github/workflows/yaml_check.yaml:4]: This doesn't have to be "build". If we rather have a different label.

Comment (grayside) [reply .github/workflows/yaml_check.yaml]: Switched to `check`

File: src/connectors/connector_cloudbuild.workflows.yaml (hunk 2 above)
Comment (grayside) [reply src/connectors/connector_cloudbuild.workflows.yaml:20]: I'm thinking use this as the basis for a docblock standard.

File: .yamllint.yaml (hunk 1 above)
Comment (grayside) [reply .yamllint.yaml:17]: SGTM

**Comment by averikitsch** [comment]
Closing in favor of #116

**Pull request closed without merging**